### PR TITLE
feat: add KongCertificate CRD

### PR DESCRIPTION
**Pull request by tao12345666333**
**What this PR does / why we need it**:

Add KongCertificate CRD

**Which issue this PR fixes**

part of https://github.com/Kong/gateway-operator/issues/581

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: api/configuration/v1alpha1/kong_certificate.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec KongCertificateSpec `json:"spec"`
+
+	// +kubebuilder:default={conditions: {{type: "Programmed", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}}
+	Status KongCertificateStatus `json:"status,omitempty"`
+}
+
+// GetKonnectStatus returns the Konnect status contained in the KongRoute status.
```
Comment (pmalek) [inline api/configuration/v1alpha1/kong_certificate.go:29]: All these Konnect related funcs should be generated with https://github.com/Kong/kubernetes-configuration/tree/main/scripts/konnect-funcs

as stated in the README: https://github.com/Kong/kubernetes-configuration/blob/d1106e28743b5c87d4adb5d36725251c4f2581ff/README.md#L52-L53

File: test/crdsvalidation/kongcertificate/testcases/required_fields.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package testcases
+
+import (
+	"github.com/samber/lo"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+var requiredFields = testCasesGroup{
```
Comment (pmalek) [inline test/crdsvalidation/kongcertificate/testcases/required_fields.go:9]: Do you mind adding a positive case here where are the required fields are specified?

File: api/configuration/v1alpha1/kong_certificate.go (hunk 1 above)
Comment (tao12345666333) [reply api/configuration/v1alpha1/kong_certificate.go]: Thank you!

File: api/configuration/v1alpha1/kong_certificate.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+// KongCertificateSpec contains the specification for the KongCertificate.
+type KongCertificateSpec struct {
+	// ControlPlaneRef references the Konnect Control Plane that this KongCertificate should be created in.
+	ControlPlaneRef        *ControlPlaneRef `json:"controlPlaneRef,omitempty"`
+	KongCertificateAPISpec `json:",inline"`
+}
+
+// KongCertificateAPISpec contains the API specification for the KongCertificate.
+type KongCertificateAPISpec struct {
```
Comment (pmalek) [inline api/configuration/v1alpha1/kong_certificate.go:37]: Aren't we missing the `ID` field in here as per https://kong-platform-api.netlify.app/konnect/control-planes-config/v2/openapi.yaml/#tag/Certificates/operation/create-certificate ?

Comment (pmalek) [reply api/configuration/v1alpha1/kong_certificate.go:37]: Hm I've just noticed now that we don't set that field when creating resources on Konnect using the SDK , e.g. 

- https://github.com/Kong/gateway-operator/blob/b449bc72d51a14c26bab4fe9bbbffefb13b93c2c/controller/konnect/ops/ops_kongconsumer.go#L337-L341
- https://github.com/Kong/gateway-operator/blob/b449bc72d51a14c26bab4fe9bbbffefb13b93c2c/controller/konnect/ops/ops_kongroute.go#L125-L146

even though the fields are in place:
- https://github.com/Kong/sdk-konnect-go/blob/077efc64e23b051ebff780fd2c05b5113463f74f/models/components/consumerinput.go#L8
- https://github.com/Kong/sdk-konnect-go/blob/077efc64e23b051ebff780fd2c05b5113463f74f/models/components/route.go#L385

Let's keep it that way for certificate as well.

**Review by pmalek — APPROVED** [review]

**Pull request merged**